txn/mgo_test.go

```diff
@@ -17,16 +21,39 @@ type MgoSuite struct {
 	session *mgo.Session
 }
 
-var mgoaddr = "127.0.0.1:50017"
+const mgoip = "127.0.0.1"
```

Comment:
The changes in this file seem unrelated to the fix. Note that most of the mgo tests use fixed ports for the tests, so we have larger problems if that was an issue.

Also, we have a test server package nowadays which could handle this logic all by itself, probably. So if we have to fix it, that'd be a better way.